test/users/npadmana/fftw/testFFTW.chpl

```diff
@@ -1,29 +1,79 @@
 
+The code computes a series of 1,2 and 3D transforms, exercising the complex <-> complex, and real<->complex
+transforms (both in and out of place). The output of the code should be a series of small numbers ( <= 10^-13);
+see testFFTW.good for example values (it is possible that your values might differ). 
 
+The data for these tests is in arr{1,2,3}d.dat. The format of these files is documented below.
+*/
+use FFTW;
+
+/* Utility function to print the maximum absolute deviation between values computed by this code, 
+   and "truth". The true values are computed using Mathematica v10 */
 proc printcmp(x, y) {
   var err = max reduce abs(x-y);
-  //writeln("----");
-  //writeln(x);
-  //writeln("**");
-  //writeln(y);
-  //writeln("----");
   writeln(err);
 }
 
+/* This is main test code, parametrized on the number of dimensions ndim. 
+  fn is the file that contains the test data.
+*/
 proc runtest(param ndim : int, fn : string) {
-  /* Read in the data and set up the domains appropriately */
   var dims : ndim*int(32); 
+
+  /* We define a number of different domains below, corresponding to complex input/output arrays,
+     real (input) -> complex (output) out-of-place arrays, and real <-> complex in-place arrays.
+
+     The domains are :
+     * D : for complex <-> complex transforms. This domain is also used for the real array in 
+           a real->complex out-of-place transform.
+     * cD : for the complex array in a real <-> complex out-of-place transform.
+     * rD : for the real array in a real <-> complex in-place transform. This includes the padding needed 
+            by the in-place transform. D is a sub-domain of this, and can be used to extract the real array, 
+            without padding.
+     * reD, imD : Utility domains that access the real/complex parts of the complex array in a real<->complex in-place transform.
+  */
   var D : domain(ndim);
-  // Define ranges here
   var rD,cD,reD,imD : domain(ndim,int,true); 
+  
+
+  /* Read in the arrays from the file below. A,B are the arrays that will be used in the FFTW calls, 
+     goodA and goodB store the true values. 
+
+     Note that goodA is all real, allowing us to reuse the same array for the real<->complex tests.
+
+     The file format is as follows (all little-endian):
+        [ndim] int32 : dimensions of the array
+        [narr] real64 : the real components of A (imaginary components are all zero). narr = N_1*N_2*..*N_ndim (product of dimensions)
+        [narr] complex128 : the components of B (stored as real-imaginary pairs).
+
+        The arrays are all stored in C (row-major) order.
+  */
   var A,B,goodA,goodB : [D] complex;
   {
     var f = open(fn,iomode.r).reader(kind=iokind.little);
+    // Read in dimensions 
     for ii in 1..ndim {
       f.read(dims(ii));
     }
-    // Note compiler does not automatically fold select on params
+    /* The code below sets the domain D. Note that the if statements are folded at compile
```

Comment:
I am increasingly considering that I misjudged this and that it wouldn't be a good use of time.  Sorry.